helpers/repo.ts

```diff
@@ -47,19 +48,82 @@ export async function downloadAndExtractRepo(
   await promises.unlink(tempFile);
 }
 
-export async function getRepoRootFolders(
+const getRepoInfo = async (owner: string, repo: string) => {
+  const repoInfoRes = await got(
+    `https://api.github.com/repos/${owner}/${repo}`,
+    {
+      responseType: "json",
+    },
+  );
+  const data = repoInfoRes.body as any;
+  return data;
+};
+
+export async function getProjectOptions(
   owner: string,
   repo: string,
-): Promise<string[]> {
-  const url = `https://api.github.com/repos/${owner}/${repo}/contents`;
+): Promise<
+  {
+    value: CommunityProjectConfig;
+    title: string;
+  }[]
+> {
+  // return community path for projects and submodules
+  const getCommunityProjectConfig = async (
+    item: any,
+  ): Promise<CommunityProjectConfig | null> => {
+    // if item is a folder, return the path with default owner, repo, and main branch
+    if (item.type === "dir")
+      return {
+        owner,
+        repo,
+        branch: "main",
+        filePath: item.path,
+      };
+
+    // check if it's a submodule (has size = 0 and different owner & repo)
+    if (item.type === "file") {
+      if (item.size !== 0) return null; // submodules have size = 0
+
+      // get owner and repo from git_url
+      const { git_url } = item;
+      const startIndex = git_url.indexOf("repos/") + 6;
+      const endIndex = git_url.indexOf("/git");
+      const ownerRepoStr = git_url.substring(startIndex, endIndex);
+      const [owner, repo] = ownerRepoStr.split("/");
```

Comment:
@thucpn this looks all a bit like it would break easily - I left a comment to use https://github.com/octokit next time if possible